src/meta/src/hummock/manager/compaction_group_manager.rs

```diff
@@ -617,67 +642,46 @@ impl HummockManager {
     ) -> Result<()> {
         // 1. Due to version compatibility, we fix some of the configuration of older versions after hummock starts.
         let current_version = &versioning_guard.current_version;
-        let all_group_ids = get_compaction_group_ids(current_version);
-        let mut configs = compaction_group_manager
-            .get_or_insert_compaction_group_configs(&all_group_ids.collect_vec())
-            .await?;
+        let all_group_ids = get_compaction_group_ids(current_version).collect_vec();
+        let default_config = compaction_group_manager.default_compaction_config();
+        let mut compaction_groups_txn = compaction_group_manager.start_compaction_groups_txn();
+        compaction_groups_txn.insert_not_exist_groups_with_config(&all_group_ids, default_config);
+
+        let compaction_groups =
+            compaction_groups_txn.try_get_compaction_group_configs(&all_group_ids);
 
         // We've already lowered the default limit for write limit in PR-12183, and to prevent older clusters from continuing to use the outdated configuration, we've introduced a new logic to rewrite it in a uniform way.
         let mut rewrite_cg_ids = vec![];
-        let mut restore_cg_to_partition_vnode: HashMap<u64, BTreeMap<u32, u32>> =
-            HashMap::default();
-        for (cg_id, compaction_group_config) in &mut configs {
+        for group in compaction_groups {
             // update write limit
+            let cg_id = group.group_id;
             let relaxed_default_write_stop_level_count = 1000;
-            if compaction_group_config
-                .compaction_config
-                .level0_sub_level_compact_level_count
+            if group.compaction_config.level0_sub_level_compact_level_count
                 == relaxed_default_write_stop_level_count
             {
-                rewrite_cg_ids.push(*cg_id);
-            }
-
-            if let Some(levels) = current_version.levels.get(cg_id) {
-                if levels.member_table_ids.len() == 1 {
-                    restore_cg_to_partition_vnode.insert(
```

Comment:
Why do we remove this code? Is it because we think that we have kept the compatibility code for a while and it is safe to remove it in the new version? If yes, how about also removing the write stop level count rewrite codes above?